### ISSUE TITLE
cmake configure error

### ISSUE THREAD

**Issue by yinweijie** [hsorby/OpenFOAM#1]
By running:
```bash
cmake -S . -B build
```
I get:
```bash
CMake Error at /home/ywj20/cmake-3.26.5-linux-x86_64/share/cmake-3.26/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find SCOTCH (missing: SCOTCH_INCLUDE_DIR)
Call Stack (most recent call first):
  /home/ywj20/cmake-3.26.5-linux-x86_64/share/cmake-3.26/Modules/FindPackageHandleStandardArgs.cmake:600 (_FPHSA_FAILURE_MESSAGE)
  cmakemodules/FindSCOTCH.cmake:274 (find_package_handle_standard_args)
  CMakeLists.txt:22 (find_package)
```
It seems that I should install SCOTCH first?

**Comment by hsorby** [comment]
That is correct.

**Comment by yinweijie** [comment]
> That is correct.

SCOTCH can be found in ThirdParty. Can I solve this problem by compiling ThirdParty package in the openfoam main project? It seems that SCOTCH_INCLUDE_DIR still not be defined after compilation of ThirdParty.

**Comment by hsorby** [comment]
I would like to think so but I didn't solve the problem that way.  I am on macOS and installed it via brew.  I haven't as yet tried installing my own build of SCOTCH to use with OpenFOAM.  
What OS are you using?

**Comment by yinweijie** [comment]
> I would like to think so but I didn't solve the problem that way. I am on macOS and installed it via brew. I haven't as yet tried installing my own build of SCOTCH to use with OpenFOAM. What OS are you using?

I use ubuntu WSL2. I'll try to intall scotch via apt install later time

**Comment by hsorby** [comment]
FWIW: I have a better build here with an updated version of OpenFOAM. https://github.com/openfoamdotcom/openfoam.

**Comment by yinweijie** [comment]
> FWIW: I have a better build here with an updated version of OpenFOAM. https://github.com/openfoamdotcom/openfoam.

that's so cool !!!

**Comment by yinweijie** [comment]
> FWIW: I have a better build here with an updated version of OpenFOAM. https://github.com/openfoamdotcom/openfoam.

Are the source codes in openfoamdotcom completely identical to official openfoam? I find some updates in src/ directory in the github page.

**Comment by hsorby** [comment]
Not quite sure what you mean but I updated to the most recent release for OpenFOAM from OpenFOAM.com.  Not the current development stage.

**Comment by yinweijie** [comment]
> Not quite sure what you mean but I updated to the most recent release for OpenFOAM from OpenFOAM.com. Not the current development stage.

![image](https://github.com/hsorby/OpenFOAM/assets/23356124/658bf3db-f659-4a5a-8689-ad6fffde98ad)

Like here, you made some updates to openfoam source code. Are you changing the code to adapt to your cmakelists or are you only migrating official openfoam source code to current standalone project ?

**Comment by hsorby** [comment]
Yes, I had to make some changes to the codebase to work with the CMake build description changes that I made.  I don't link together the include directories like the wmake build system does. 

I didn't make any *real* code changes though, just slight modifications for creating build instructions with CMake.

**Comment by yinweijie** [comment]
> Yes, I had to make some changes to the codebase to work with the CMake build description changes that I made. I don't link together the include directories like the wmake build system does.
> 
> I didn't make any _real_ code changes though, just slight modifications for creating build instructions with CMake.

Thanks, that's already good enough. This project seems support for macos well from your project descriptions, let me try with linux os.

**Comment by hsorby** [comment]
I have managed to build on a small HPC Linux environment, that worked quite well.  I think it is a CentOS environment from memory.

**Comment by yinweijie** [comment]
> I have managed to build on a small HPC Linux environment, that worked quite well. I think it is a CentOS environment from memory.

I passed compilation on my ubuntu 22 laptop with some minor change to cmakelists.

**Comment by hsorby** [comment]
Nice, feel free to share the changes on that repo.

**Comment by yinweijie** [comment]
> Nice, feel free to share the changes on that repo.

https://github.com/openfoamdotcom/openfoam/pull/1
